src/file_lines.rs

```diff
@@ -151,6 +151,21 @@ impl<'a> iter::Iterator for Files<'a> {
     }
 }
 
+fn canonicalize_path_string(s: &str) -> String {
+    use std::path;
+    use std::fs;
+
+    match fs::canonicalize(path::PathBuf::from(s)) {
+        Ok(canonicalized) => {
+            match canonicalized.to_str() {
+                Some(c) => c.to_string(),
+                _ => String::new(),
+            }
+        }
+        _ => String::new(),
```

Comment:
These two `String::new()`s worry me - it seems like if we fail, we should bail out rather than silently return an empty string. Maybe this method could return a `Result` or it could `panic!`, in any case we should warn the user something went wrong.